trussworks/react-uswds
# fix: a11y omnibus package

# Summary

Fixes a bunch of a11y issues that [Playwright](https://playwright.dev/docs/accessibility-testing) surfaced (322 of 553 tests failed):

**Source changes:**
- Required aria-hidden or aria-label or aria-labelledby on all `Icons`, defaulting to the former being true in storybook. Removed part of a test to this end.
- Changed `CardHeader` element from `header` to `div` to avoid [a11y issues](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/header#accessibility_concerns) that Storybook helpfully demonstrated, and it's how [USWDS implements](https://designsystem.digital.gov/components/card/#component-code) it anyway (probably for the same reason). This might be breaking in case anyone used css to target the rendered semantic `header` element. If so, please mark as breaking.
- Hid `Pagination`, `SocialLinks`, and `SearchButton`'s decorative svgs from screen readers
- Replaced presentation role with aria-label to fix a11y issue, and per USWDS implementation
- Used aria-controls on `LanguageSelector` only if specified, and for this component only uses it when appropriate

**Storybook-only changes:**
- Added labels to all `DatePicker`, `InputPrefix`, `InputSufffix`, `RangeInput`, `Select`, `Textarea`, and `TextInput` stories to fix a11y warnings there, and tidied up a bit
- Changed `Tag` story colors to meet contrast requirements
- Hid `IconList` icons from screen readers
- Changed `Footer`'s `h3`s to `p`s since class already handled styling and use of a heading seemed unwarranted
- Tidied `Pagination`'s stories
- Disabled 'skip-link' rule for `ExtendedNav` due to false-positive in Storybook

## Related Issues or PRs

Resolves #2811 
Resolves #2809

## How To Test

If you have Playwright installed locally, run `yarn test-storybook`. Otherwise, check the Accessibility tab of all the stories

### Screenshots (optional)
